benchmarks/syscall_analysis.sh

```
@@ -0,0 +1,326 @@
+
+for i in $(seq 1 "$NUM_FILES"); do
+    dd if=/dev/urandom of="$TEST_DIR_SRC/file${i}.bin" bs=1M count="$FILE_SIZE_MB" 2>/dev/null
+done
+
+echo "✓ Created $NUM_FILES files × ${FILE_SIZE_MB}MB = $((NUM_FILES * FILE_SIZE_MB))MB"
+echo ""
+
+# Run arsync with full syscall trace
+echo "Running arsync with strace..."
+strace -c -f -o "$TRACE_SUMMARY" "$ARSYNC_BIN" "$TEST_DIR_SRC" "$TEST_DIR_DST" -a 2>&1 > /dev/null || true
+strace -e trace=all -f -o "$TRACE_RAW" "$ARSYNC_BIN" "$TEST_DIR_SRC" "$TEST_DIR_DST"2 -a 2>&1 > /dev/null || true
```

Shellcheck (0.11.0)

[warning] To redirect stdout+stderr, 2>&1 must be last (or use '{ cmd > file; } 2>&1' to clarify).

(SC2069)